[BUG] test_log_transform test fails on 32-bit systems

On 32-bit (i586) architecture, the test_log_transform test fails with the following report (using `pytest`), but passes fine on 64-bit architectures. Full log attached below in case it helps.

Boost-histogram version: 1.0.2
Python version: 3.9

```
[  419s] =================================== FAILURES ===================================
[  419s] ________________________ TestRegular.test_log_transform ________________________
[  419s] 
[  419s] self = <test_axis.TestRegular object at 0xf4c4bd60>
[  419s] 
[  419s]     def test_log_transform(self):
[  419s]         a = bh.axis.Regular(2, 1e0, 1e2, transform=bh.axis.transform.log)
[  419s]     
[  419s]         assert a.index(-1) == 2
[  419s]         assert a.index(0.99) == -1
[  419s]         assert a.index(1.0) == 0
[  419s]         assert a.index(9.99) == 0
[  419s] >       assert a.index(10.0) == 1
[  419s] E       assert 0 == 1
[  419s] E         +0
[  419s] E         -1
[  419s] 
[  419s] tests/test_axis.py:328: AssertionError
[  419s] =========================== short test summary info ============================
[  419s] FAILED tests/test_axis.py::TestRegular::test_log_transform - assert 0 == 1
```

Full build log: [_log.zip](https://github.com/scikit-hep/boost-histogram/files/6757786/_log.zip)

--- a/tests/test_axis.py
+++ b/tests/test_axis.py
@@ -325,12 +325,12 @@ class TestRegular(Axis):
 
         assert a.index(-1) == 2
         assert a.index(0.99) == -1
-        assert a.index(1.0) == 0
+        assert a.index(1.001) == 0
         assert a.index(9.99) == 0
-        assert a.index(10.0) == 1
+        assert a.index(10.01) == 1
         assert a.index(99.9) == 1
-        assert a.index(100) == 2
-        assert a.index(1000) == 2
+        assert a.index(100.01) == 2
+        assert a.index(1000.1) == 2
 
         assert a.bin(0)[0] == approx(1e0)
         assert a.bin(1)[0] == approx(1e1)

